woocommerce/woocommerce-ios
# Variations: Create Variable Product

closes #3802 #3050 

# Why

Since now we can add variations to a variable product, it's time to enable the ability to create a variable product(with variations) from scratch.

# How

Unfortunately, this wasn't as straightforward. 😓 

In order to add attributes or variations to a new product, we need to create that product as a draft first. This is done in `EditAttributeOptionsViewModel` right after updating the product attributes. Here we check if the product exists remotely to know if we should update the product or create the product as a draft first.

That draft creation has some cascade consequences like:

- The `resultsController` in `ProductVariationViewController` is now invalid since the product has a different ID now. This is fixed by recreating and reconfiguring the `resultsController` with using the new product information.

- The `formType` of `ProductForm` screen and `ProductVariation` screen has to change from `.add` to `.edit` in order to reflect properly the current state of the product. This is done both in `ProductFormViewModel` and in `ProductVariationsViewModel`


# Demo

https://user-images.githubusercontent.com/562080/111397454-589c0f80-868f-11eb-8d82-ad31aefe816f.mov

# Testing Steps
- Open the app and go to the product tab
- Tap on the "+" button and choose variable product
- See that the product type is now visible in the add product form.
- Create a product with attributes and variations as you see fit.
- See that the product form is now in edit mode.
- See that the product has the draft status
- Update the product to published status if you wish.


Update release notes:
